### PR TITLE
Add jenkins_cli Exploit

### PR DESCRIPTION
**Pull request by JD2344**

### CODE REVIEW

**Comment by JD2344** [comment]
Should be good to go. 

Tested, runs as user and dynamic port.

File: modules/vulnerabilities/unix/http/jenkins_cli/secgen_metadata.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  <read_fact>port</read_fact>
+  <read_fact>strings_to_leak</read_fact>
+  <read_fact>leaked_filenames</read_fact>
+
+  <default_input into="port">
+    <generator module_path=".*/random_unregistered_port" />
+  </default_input>
+
+  <default_input into="strings_to_leak">
+    <generator type="flag_generator" />
```
Comment (cliffe) [inline modules/vulnerabilities/unix/http/jenkins_cli/secgen_metadata.xml:34]: Minor, but except where it's specifically a CTF puzzle challenge, we avoid having flags as the default

File: modules/vulnerabilities/unix/http/jenkins_cli/secgen_metadata.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  </default_input>
+
+  <default_input into="leaked_filenames">
+    <value>flag</value>
+  </default_input>
+
+  <default_input into="leaked_username">
```
Comment (cliffe) [inline modules/vulnerabilities/unix/http/jenkins_cli/secgen_metadata.xml:41]: Is this username leaked to the attacker after exploitation? If not this should change.

**Review by cliffe — APPROVED** [review]
I'm going to merge, but please get back to me on whether the username is leaked to the attacker, if not the parameter name needs to change.

**Pull request merged**